code/modules/clothing/shoes/magboots.dm

```diff
@@ -14,6 +14,39 @@
 	var/stomp_hit = "crushes"
 	var/anchoring_system_examine = "Its mag-pulse traction system appears to be"
 
+	var/obj/item/clothing/shoes/stored_shoes = null	//Shoe holder
+
+/obj/item/clothing/shoes/magboots/mob_can_equip(mob/living/carbon/human/user)
+	var/mob/living/carbon/human/H = user
+	if(!istype(H) || stored_shoes)
+		return ..()
+	if(H.shoes)
+		stored_shoes = H.shoes
+		if(istype(stored_shoes, /obj/item/clothing/shoes/magboots))
```

Comment:
instead of a direct type check, why not a var for overshoe?

actually, even better, why not just check to make sure its w_class is smaller than our w_class and make magboots 1 size larger than other shoes? that would be the most immersive option

Reply:
That would make mags too big for bags, but i like it way more than adding a var